[LynetteLeeSiYing] iP

LynetteLeeSiYing opened the pull request:
# Socchat :cat:
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

Socchat frees your mind of having to remember things you need to do. It's,
- text-based
- **_Fast_** to use

## Getting started
1. Download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/)
2. Double-click it
3. Add your tasks
4. Let Socchat manage your tasks for you!

## Features
- [x] Manage various types of tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Add greeting and exit

Commit:
echo messages

Commit:
Add and list tasks

Commit:
Mark as done and not done

Commit:
added three different tasks

Commit:
Automated Text UI Testing

Commit:
rearrange and handle errors

Commit:
added delete and change use enums to replace if-else

Commit:
Save tasks into a file

Commit:
Merge branch 'branch-Level-7'

Commit:
Change date format

Commit:
Debug exit command

Commit:
Merge branch 'branch-Level-8'

Commit:
Create more OOP

Commit:
Implement OOP

Commit:
Implement more OOp

Commit:
Organize into packages

Commit:
Merge branch 'A-Packages'

Commit:
Merge remote-tracking branch 'origin/add-gradle-support'

Commit:
Add Junit Tests

Commit:
Add test

Commit:
Merge branch 'A-JUnit'

Commit:
Add JavaDoc

Commit:
Merge branch 'A-JavaDoc'

Commit:
Follow coding standard

Commit:
Merge branch 'A-CodingStandard'

Commit:
Implement find

Commit:
Merge branch 'Level-9'

seah-minlong commented on src/main/java/socchat/Storage.java:
```diff
@@ -0,0 +1,108 @@
+package socchat;
+
+import socchat.task.Task;
+import socchat.task.todo.*;
+import socchat.task.deadline.*;
+import socchat.task.event.*;
```
Perhaps the imported classes should be listed explicitly as per the coding standard?

seah-minlong commented on src/main/java/socchat/Socchat.java:
```diff
@@ -0,0 +1,140 @@
+    public void run() {
+        greet();
+
+        chatLoop:
+        while (true) {
+            String[] strToken;
+            try{
+                String input = scanner.next();
+                Command command = getCommand(input);
+                System.out.print("> ");
+                switch (command) {
+                case BYE:
+                    exit();
+                    break chatLoop;
```
I like how you used this `chatLoop:` notation! 💯

seah-minlong commented on src/main/java/socchat/Storage.java:
```diff
@@ -0,0 +1,108 @@
+
+                String done = strToken[1].trim();
+
+                String des = strToken[2].trim();
```
Perhaps a better name can be used here to explain what this `des` string is?

seah-minlong commented on src/main/java/socchat/TaskList.java:
```diff
@@ -0,0 +1,139 @@
+
+    public void delete(String indexString) throws SocchatException {
+        try {
+//            String taskIndexString = scanner.nextLine().trim();
```
Perhaps the comment here should be indented relative to their position in the code as per the coding standards? 

```
while (true) {
    // Do something
    something();
}
```

seah-minlong commented on src/main/java/socchat/task/Task.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    public abstract String toSave();
+    //...
```
Perhaps this comment without anything should be deleted?

seah-minlong commented on src/main/java/socchat/Storage.java:
```diff
@@ -0,0 +1,108 @@
+                tasks.add(t);
+            }
+        } catch (FileNotFoundException e) {
+              throw  new SocchatException("Storage file not found");
```
Perhaps there should only be 1 spacing here between throw and new?

seah-minlong reviewed (commented):
Overall, I think this is a great start! 😄 
Perhaps the `print` statements can be shifted to the `Ui` class to make the code more organised? I also noticed a few coding standard violations, I think using Gradle's checkstyle would be useful to catch these mistakes!

samuelthen commented on src/main/java/socchat/Storage.java:
```diff
@@ -0,0 +1,108 @@
+
+                String des = strToken[2].trim();
+
+                Task t;
```
More descriptive field names for `des` and `t` will help!

samuelthen commented on src/main/java/socchat/Storage.java:
```diff
@@ -0,0 +1,108 @@
+package socchat;
+
+import socchat.task.Task;
+import socchat.task.todo.*;
```
Please list the imported classes explicitly.

samuelthen commented on src/test/java/socchat/ParserTest.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
```
Should an empty line be added to make it more organized?

samuelthen reviewed (commented):
### Overall Comments

I like how your code is structured in such a way that is easy to read and organized. There are just a few coding standard errors to edit!

**Well done**!

Commit:
Check style

Commit:
Merge branch 'A-CheckStyle'

Commit:
Add GUI to chatbot

Commit:
Merge branch 'Level-10'

Commit:
Apply varargs

Commit:
Merge branch 'A-Varargs'

Commit:
Add assertions

LynetteLeeSiYing committed:
Merge pull request #1 from LynetteLeeSiYing/branch-A-Assertions

Add assertions

Commit:
Improve code quality.

- TaskList class: contains some common behaviors while processing storage for different tasks.
- TaskList class: Methods are not adhering with logical structure.
- Storage class: Else branch is used for task type = 'D'

The common behaviors cause code duplication, lines without logical structure
will affect clarity.The 'else' branch should not be used when the condition can be specified.

- Let's pull up behaviors common to all tasks into a new method named addingTaskAcknowledgement.
- Let's add some black lines to separate groups of related statements
- Let's use the 'else' branch for catch error and move the 'D' task into if-else branch.

LynetteLeeSiYing committed:
Merge pull request #2 from LynetteLeeSiYing/branch-A-CodeQuality

Improve code quality.

Commit:
New feature: tagging

Tagging allows users to categorize their task when
task getting more in future.

'-tag ' option allows users to input their tags.
In future implementation, we will include a way to
filter out the tags so that users can view all tasks
in a specific tag.

Let's,
- search for the keyword '-tag'
- capture and store users' tags together with the task

Commit:
Merge branch 'C-tagging'

Commit:
Tweak GUI

Commit:
Add personality

Commit:
Add error-handling for start/end date

Commit:
Merge branch 'A-MoreErrorHandling'

Commit:
Add user guide

Commit:
Modify user guide

Commit:
Modify user guide

Commit:
Improve code quality and add more javadocs with AI

Commit:
Edit user guide

Commit:
Resolve index bug
